### PR TITLE
mds: obsoleting 'mds_cache_size'

### PR DESCRIPTION
**Pull request by ajarr**
Remove last bits of support for 'mds_cache_size'.
'mds_cache_memory_limit' is preferred.

Fixes: https://tracker.ceph.com/issues/41951
Signed-off-by: Ramana Raja <rraja@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/rados/configuration/ceph-conf.rst [hunk 1]
```diff
@@ -144,7 +144,7 @@ These sections include:
               the Ceph Storage Cluster, and override the same setting in
               ``global``.
 
-:Example: ``mds_cache_size = 10G``
+:Example: ``mds_cache_memory_limit = 1073741824``
```
Comment (batrick) [inline doc/rados/configuration/ceph-conf.rst]: `10G` works and is more readable.

**Comment by batrick** [comment]
> Did you also check qa is not using the old setting?

?

Also, it's best to add the releaes note in a random location in the file to reduce the likelihood of conflicts. Please rebase.

File: doc/cephfs/mds-config-ref.rst [hunk 2]
```diff
@@ -13,7 +13,6 @@
 ``mds cache memory limit``
 
 :Description: The memory limit the MDS should enforce for its cache.
-              Administrators should use this instead of ``mds cache size``.
 :Type:  64-bit Integer Unsigned
 :Default: ``1073741824``
```
Comment (batrick) [inline doc/cephfs/mds-config-ref.rst]: Update this to `1G`

File: qa/tasks/cephfs/test_client_limits.py [hunk 3]
```diff
@@ -38,15 +38,16 @@ def _test_client_pin(self, use_subdir, open_files):
         :param use_subdir: whether to put test files in a subdir or use root
         """
 
-        cache_size = open_files/2
+        cache_memory_limit = open_files/2 * 2000
```
Comment (batrick) [inline qa/tasks/cephfs/test_client_limits.py]: I think you can just set this to a really low number, like "1M". Have you tried running it?

**Comment by batrick** [comment]
> > Did you also check qa is not using the old setting?
> 
> ?

Nevermind, I see you've made changes to `qa/` now.

File: qa/tasks/cephfs/test_client_limits.py (hunk 3 above)
Comment (ajarr) [reply qa/tasks/cephfs/test_client_limits.py]: Done

File: doc/cephfs/mds-config-ref.rst (hunk 2 above)
Comment (ajarr) [reply doc/cephfs/mds-config-ref.rst]: Done

**Comment by ajarr** [comment]
> Also, it's best to add the releaes note in a random location in the file to reduce the likelihood of conflicts. Please rebase.

Nice idea. Done!

**Pull request merged**